java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/MultiCryptoProviderDecryptionTest.java

```diff
@@ -0,0 +1,279 @@
+        validateEncryptedFileWithHeaderValues(cipherDetails);
+
+    }
+
+    public void decryptAES192CTRNoPadding() throws IOException {
+
+        final SupportedCipherDetails cipherDetails = CIPHERS_TO_TEST.get(4);
+        validateEncryptedFileWithHeaderValues(cipherDetails);
+
+    }
+
+    public void decryptAES256GCTRNoPadding() throws IOException {
+
+        final SupportedCipherDetails cipherDetails = CIPHERS_TO_TEST.get(5);
+        validateEncryptedFileWithHeaderValues(cipherDetails);
+
+    }
+
+    public void decryptAES128CBCPKCS5Padding() throws IOException {
+
+        final SupportedCipherDetails cipherDetails = CIPHERS_TO_TEST.get(6);
+        validateEncryptedFileWithHeaderValues(cipherDetails);
+
+    }
+
+    public void decryptAES196CBCPKCS5Padding() throws IOException {
+
+        final SupportedCipherDetails cipherDetails = CIPHERS_TO_TEST.get(7);
+        validateEncryptedFileWithHeaderValues(cipherDetails);
+
+    }
+
+    public void decryptAES256CBCPKCS5Padding() throws IOException {
+
+        final SupportedCipherDetails cipherDetails = CIPHERS_TO_TEST.get(8);
+        validateEncryptedFileWithHeaderValues(cipherDetails);
+
+    }
+
+    /* Test Utility methods used in all unit-tests. */
+
+    private void validateEncryptedFileWithHeaderValues(SupportedCipherDetails cipherDetails) throws IOException {
+        final String cipherName = cipherDetails.getCipherId().replaceAll("/","-");
+        final SecretKey secretKey = generateSecretKey(cipherDetails);
+        final String filePath = "test-data/ciphertext/" + cipherName + ".ciphertext";
+        final URL encryptedFileURL = getClass().getClassLoader().getResource(filePath);
+
+        MantaHttpHeaders getResponseHttpHeaders = new MantaHttpHeaders();
+        populateHeaderValuesFromResources(getResponseHttpHeaders,cipherDetails);
+
+
+        final long cipherTextSize = getResponseHttpHeaders.getContentLength();
+
+        try (InputStream encryptedStream = encryptedFileURL.openStream();
+             BoundedInputStream bin = new BoundedInputStream(encryptedStream, cipherTextSize)) {
+
+            final byte[] actualBytes = IOUtils.toByteArray(bin);
+
+            try (final MantaEncryptedObjectInputStream decryptingStream = new MantaEncryptedObjectInputStream(
+                    new MantaObjectInputStream(
+                            new MantaObjectResponse(filePath, getResponseHttpHeaders),
+                            Mockito.mock(CloseableHttpResponse.class),
+                            new EofSensorInputStream(
+                                    new ByteArrayInputStream(actualBytes),
+                                    null)),
+                    cipherDetails,
+                    secretKey,
+                    true)) {
+
+                ByteArrayOutputStream decrypted = new ByteArrayOutputStream();
+                IOUtils.copy(decryptingStream, decrypted);
+                decryptingStream.close();
```

Comment:
You don't need to close this because it is being closed with the try with resources operation.

Reply:
Removed unnecessary close statement.